src/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkActionsSaveToListsButton.tsx

```diff
@@ -0,0 +1,49 @@
+import { ContextModule } from "@artsy/cohesion"
+import { SaveUtilButton } from "Apps/Artwork/Components/ArtworkImageBrowser/SaveUtilButton"
+import { useSaveArtworkToLists } from "Components/Artwork/SaveButton/useSaveArtworkToLists"
+import { FC } from "react"
+import { createFragmentContainer, graphql } from "react-relay"
+import { ArtworkActionsSaveToListsButton_artwork$data } from "__generated__/ArtworkActionsSaveToListsButton_artwork.graphql"
+
+interface ArtworkActionsSaveToListsButtonProps {
+  artwork: ArtworkActionsSaveToListsButton_artwork$data
+}
+
+export const ArtworkActionsSaveToListsButton: FC<ArtworkActionsSaveToListsButtonProps> = ({
+  artwork,
+}) => {
+  const { isSaved, onSave } = useSaveArtworkToLists({
+    artwork,
+    contextModule: ContextModule.artworkImage,
+    customListsCount: artwork.customCollections?.totalCount ?? 0,
+    isSaved: !!artwork.isSaved,
+  })
+
+  return <SaveUtilButton isSaved={isSaved} onClick={onSave} />
+}
+
+export const ArtworkActionsSaveToListsButtonFragmentContainer = createFragmentContainer(
```

Comment:
hey @dimatretyak, just wanted to share my 2 cents, even though I don't know how I would do it better. My concern is that some data, like `title` for example is requested from here, but it is not explicitly used in this file, but in the custom hook. I am wondering if there is a way to declare which data is needed for custom hook functionality in `useSaveArtworkToLists` file itself. This way the code will become less error-prone I believe

Reply:
> declare which data is needed

Here it describes what data is needed for `useSaveArtworkToLists` and if the user does not pass some data, then this will be reported (see screenshot).

Yes, I understand that this is not the best approach. Ideally, we should query this data in the hook itself (for example, using `useFragmentContainer`), but so far relay hooks are not used in Force. It is for this reason that a [second approach](https://github.com/artsy/force/pull/12054#discussion_r1132347013) has been added that solves this problem.
<img width="1304" alt="image" src="https://user-images.githubusercontent.com/3513494/225580917-f2ecf8c8-57fe-4510-87cb-dcafb628c70f.png">